sklearn/mixture/bayesianmixture.py

```diff
@@ -0,0 +1,938 @@
+                               np.dot(chol_sol.T, chol_sol))
+        temp4 = -.5 * self.n_components * self.lambda_nu_ * trace_W0inv_W
+        return temp_mu + temp3 + temp4
+
+    def _estimate_p_mu_lambda_diag(self):
+        temp1 = self._log_lambda - \
+            self.n_features * self.mu_beta_prior / self.mu_beta_
+        diff = self.mu_m_ - self.mu_m_prior
+        temp2 = self.mu_beta_prior * self.lambda_nu_ * \
+            np.sum(np.square(diff) / self.lambda_inv_W_, axis=1)
+        temp_mu = .5 * np.sum(temp1 - temp2) + \
+            self._log_gaussian_norm_beta_prior
+
+        temp3 = self.n_components * self._log_gamma_norm_W_nu_prior + \
+            (self.lambda_nu_prior - 1) * np.sum(self._log_lambda)
+        temp4 = np.sum(- self.lambda_nu_ * np.sum(self.lambda_inv_W_prior /
+                                                  self.lambda_inv_W_, axis=1))
+        return temp_mu + temp3 + temp4
+
+    def _estimate_p_mu_lambda_spherical(self):
+        temp1 = self._log_lambda - \
+            self.n_features * self.mu_beta_prior / self.mu_beta_
+        diff = self.mu_m_ - self.mu_m_prior
+        temp2 = self.mu_beta_prior * self.lambda_nu_ / self.lambda_inv_W_ * \
+            np.sum(np.square(diff), axis=1)
+        temp_mu = .5 * np.sum(temp1 - temp2) + \
+            self._log_gaussian_norm_beta_prior
+
+        temp3 = self.n_components * self._log_gamma_norm_W_nu_prior + \
+            (self.lambda_nu_prior - 1) * np.sum(self._log_lambda)
+        temp4 = np.sum(- self.lambda_nu_ * self.lambda_inv_W_prior
+                       / self.lambda_inv_W_)
+        return temp_mu + temp3 + temp4
+
+    def _estimate_q_Z(self, resp):
+        """Equation 7.10
+        """
+        resp = resp[resp > 10 * EPS]
+        return np.sum(resp * np.log(resp))
+
+    def _estimate_q_weight(self):
+        """Equation 7.11
+        """
+        return np.sum((self.weight_alpha_ - 1) * self._log_pi) + \
+            _log_dirichlet_norm(self.weight_alpha_)
+
+    def _estimate_q_mu_lambda(self):
+        if self.precision_type == 'full':
+            return self._estimate_q_mu_lambda_full()
+        elif self.precision_type == 'tied':
+            return self._estimate_q_mu_lambda_tied()
+        elif self.precision_type == 'diag':
+            return self._estimate_q_mu_lambda_diag()
+        elif self.precision_type == 'spherical':
+            return self._estimate_q_mu_lambda_diag()
+
+    def _estimate_q_mu_lambda_full(self):
+        wishart_entropy = np.empty(self.n_components)
+        for k in range(self.n_components):
+            wishart_entropy[k] = _wishart_entropy(
+                self.n_features, self.lambda_nu_[k], self.lambda_inv_W_[k],
+                self._log_lambda[k])
+        return np.sum(.5 * self._log_lambda +
+                      .5 * self.n_features * np.log(self.mu_beta_ / (2 * np.pi))
+                      - .5 * self.n_features
+                      - wishart_entropy)
```

Comment:
I have a question about PEP8 style. PEP8 said about these lines ```W503 line break before binary operator```. But I think putting the ```-``` operator to the previous line might cause people who would like to check the function against the math equation to omit the minus sign. Placing ```-``` sign in the front of the line would make it noticeable.  @ogrisel